sys/include/net/ng_netapi.h

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_netapi Generic network module interface
+ * @ingroup     net
+ * @brief       Generic interface for IPC communication between network modules
+ *
+ * @details     The idea of this interface is that it provides every network
+ *              module with a basic set of commands to communicate with its
+ *              neighboring modules. In this model every module runs in its own
+ *              thread and communication is done using netapi.
+ *
+ *                     +-----+  +-----+   +-----+
+ *                     | App |  | App |   | App |
+ *                     +-----+  +-----+   +-----+
+ *                        ^        ^         ^
+ *                      netapi   netapi    netapi
+ *                        v        v         v
+ *                    +--------------------------+
+ *                    | Socket | Socket | Socket |
+ *                    |      Transport layer     |
+ *                    +--------------------------+
+ *                                 ^
+ *                               netapi
+ *                                 v
+ *                    +--------------------------+
+ *                    |      Network layer       |
+ *                    +--------------------------+
+ *                       ^         ^         ^
+ *                     netapi    netapi    netapi
+ *                       v         v         v
+ *                   +-------+ +-------+ +-------+
+ *                   |  MAC  | |  MAC  | |  MAC  |
+ *                   +-------+ +-------+ +-------+
+ *                   | Radio | | Radio | | Radio |
+ *                   +-------+ +-------+ +-------+
+ *
+ * @{
+ *
+ * @file
+ * @brief       Generic interface to communicate with network modules
+ *
+ * @author      Martine Lenders <mlenders@inf.fu-berlin.de>
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef NG_NETAPI_H_
+#define NG_NETAPI_H_
+
+#include "kernel.h"
+#include "thread.h"
+#include "ng_netconf.h"
+#include "ng_pktsnip.h"
+
+
+/**
+ * @brief   Message type for passing data up the network stack
+ */
+#define NG_NETAPI_MSG_TYPE_RCV          (0x0201)
```

Comment:
Agree with @authmillenon, using an enum here would be contrary to what we use in other RIOT code (also I don't quite see why it saves the hassle).

After thinking about it, I am also against using `#ifdefs` here. Otherwise people start to re-assign message types anywhere in there code and then it becomes almost impossible to track the message IDs actually used.